src/main/java/com/autotune/analyzer/workerimpl/BulkJobManager.java

```diff
@@ -98,176 +100,271 @@ public static List<String> appendExperiments(List<String> allExperiments, String
+                            } catch (Exception e) {
+                                e.printStackTrace();
+                                recommendationData.moveToFailed(experiment_name);
+                            }
+                        });
+                    }
+                }
+            }
+        } catch (Exception e) {
+            LOGGER.error(e.getMessage());
+            e.printStackTrace();
+            jobStatusMap.get(jobID).setStatus("FAILED");
+        }
+    }
+
 
+    Map<String, CreateExperimentAPIObject> getExperimentMap(DataSourceMetadataInfo metadataInfo) throws IOException {
+        Map<String, CreateExperimentAPIObject> createExperimentAPIObjectMap = new HashMap<>();
+        Collection<DataSource> dataSourceCollection = metadataInfo.getDataSourceHashMap().values();
+        for (DataSource ds : dataSourceCollection) {
+            HashMap<String, DataSourceCluster> clusterHashMap = ds.getDataSourceClusterHashMap();
+            for (DataSourceCluster dsc : clusterHashMap.values()) {
+                HashMap<String, DataSourceNamespace> namespaceHashMap = dsc.getDataSourceNamespaceHashMap();
+                for (DataSourceNamespace namespace : namespaceHashMap.values()) {
+                    HashMap<String, DataSourceWorkload> dataSourceWorkloadHashMap = namespace.getDataSourceWorkloadHashMap();
+                    if (dataSourceWorkloadHashMap != null) {
+                        for (DataSourceWorkload dsw : dataSourceWorkloadHashMap.values()) {
+                            HashMap<String, DataSourceContainer> dataSourceContainerHashMap = dsw.getDataSourceContainerHashMap();
+                            if (dataSourceContainerHashMap != null) {
+                                for (DataSourceContainer dc : dataSourceContainerHashMap.values()) {
+                                    // Experiment name - dynamically constructed
+                                    String experiment_name = this.bulkInput.getDatasource() + "|" + dsc.getDataSourceClusterName() + "|"
+                                            + namespace.getDataSourceNamespaceName() + "|" + dsw.getDataSourceWorkloadName() + "("
+                                            + dsw.getDataSourceWorkloadType() + ")" + "|" + dc.getDataSourceContainerName();
+                                    // create JSON to be passed in the createExperimentAPI
+                                    String createExpJSON = prepareCreateExperimentJSONInput(dc, dsc, dsw, namespace,
+                                            experiment_name, this.bulkInput.getDatasource());
+                                    // send request to createExperiment API for experiment creation
+                                    int responseCode = sendCreateExperimentRequest(createExpJSON);
+                                    // if the experiment is successfully created, add it in the map
+                                    if (responseCode == HttpURLConnection.HTTP_CREATED) {
```

Comment:
If experiments already created then also Please add it to 
createExperimentAPIObjectMap.put(jsonObject.get("experiment_name").getAsString(), `createExperimentAPIObject);`




avoid Failed one like JOB experiments etc